src/goto-programs/remove_virtual_functions.cpp

```diff
@@ -267,6 +281,49 @@ void remove_virtual_functionst::get_child_functions_rec(
   {
     if(!visited.insert(child).second)
       continue;
+
+    exprt method=get_method(child, component_name);
+    dispatch_table_entryt function(child);
+    if(method.is_not_nil())
+    {
+      function.symbol_expr=to_symbol_expr(method);
+      function.symbol_expr.set(ID_C_class, child);
+    }
+    else
+    {
+      function.symbol_expr=last_method_defn;
+    }
+    functions.push_back(function);
+
+    if(recurse)
+    {
+      get_child_functions(
```

Comment:
Why would you ever want to not recurse over child types?

Reply:
I had abused this to check only direct parents, but this is now the recursive `get_parent_function` method (and probably soon the one you pointed out)